### ISSUE TITLE
[Q] Marlin Mode Fails to Work SKR Mini E3 V3 board and TFT24 v1.1

### ISSUE THREAD

**Issue by helakejr** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2749]
I upgraded the SKR Mini E3 V3 board and TFT24 v1.1 display at the same time and the Touch Mode screen works fine however when selecting Marlin Mode, the screen remains black except for the "Marlin Mode" title. After much testing, I discovered both the Marlin and touch screen modes work if I connect the serial cable to the SWD port on the TFT24. 

Everything works fine on the Marlin Mode in Touch Mode I get "No Printer Attached" message on the touch screen. 

Marlin Configuration configured per :
#define SERIAL_PORT 2		// Host port
#define BAUDRATE 115200

#define SERIAL_PORT_2 -1
//#define BAUDRATE_2 250000   

In summary, when I connect:

BTT SKR Mini E3 V3	TFT24 v1.1
10-pin EXP1		to	TFT24 EXP1/EXP2 split cable
5-pin serial	to	TFT port	Touch mode works, Marlin Mode blank Screen
5-pin serial	to	SWD port	Touch mode does not connect, Marlin Mode works

Test Setup:
10-pin EXP1	to	TFT24 EXP1/EXP2 split cable
5-pin serial	to	TFT port	Touch mode works, Marlin Mode blank Screen
SKR Mini E3 V3 board jumpered to provide +5V - Touch mode works, Marlin Mode blank Screen
SKR Mini E3 V3 board external +5V  - Touch mode works, Marlin Mode blank Screen

I have replaced both: BTT SKR Mini E3 V3, and TFT24 V1.1 with same results.

**Comment by radek8** [comment]
Never connect TFT and SKR via the SWD connector. You risk destroying the pins that use 3.3V logic. The TFT and SWD connectors have the power pins reversed.

The problem will be in the connection of your split cable
Can you send the wiring diagram of your cable?

**Comment by helakejr** [comment]
This cable was provided by BIQU with the TFT24 v1.1.

![image](https://user-images.githubusercontent.com/751406/226116482-27f015a4-d6f2-4145-ab13-06791639bc30.png)
![Split Wiring](https://user-images.githubusercontent.com/751406/226116528-ee4ee8d7-5539-47f0-b9d8-81cc8a34e928.jpg)

**Comment by radek8** [comment]
The picture is correct. Check if the cable is connected as well as the picture..
Check the light blue lines.

**Comment by helakejr** [comment]
I checked the cable. It is built correctly. I replaced the SKR mini e3
board and the TFT24. The TFT24 had the same problem. And the cable was
built the same way.

**Comment by radek8** [comment]
Send photos of the cable plugged in.

**Comment by helakejr** [comment]
With only the 10-pin split cable connected - the TFT24 reset - resets both. However, still Marlin Mode header with blank screen.

**Comment by radek8** [comment]
I have a question.
1. If you connect only the serial cable (5-pin), does the touch mode work and can the printer be controlled?
2. If you connect only a 10-pin cable, does the Marlin mode work for you?
3. If you connect both cables, the Marlin mode does not appear, but the touch mode controls the printer?

**Comment by helakejr** [comment]
![PXL_20230318_162919504](https://user-images.githubusercontent.com/751406/226119471-628da0ed-caaa-4393-a078-e6bfc6d4a4da.jpg)

1. Yes, with only the 5-pin connected the touch mode works and the printer can be controlled
2. No, Marlin Mode header blank screen
3. Yes, the touch mode works and controls the printer, Marlin Mode header only

![PXL_20230318_163850120](https://user-images.githubusercontent.com/751406/226120660-308e91b1-3c5c-4963-a64e-e970943a5970.jpg)
![PXL_20230318_163906924](https://user-images.githubusercontent.com/751406/226120672-d507a28a-b76c-4d2e-a9a5-b2102f14af5d.jpg)

**Comment by helakejr** [comment]
Question:
What serial port does the SWD port use?

**Comment by radek8** [comment]
The SWD connector is not a serial port!
Compiling your own Marlin?
Send configuration files.

**Comment by helakejr** [comment]
I used the prebuilt with the same results.

Here is my build.
[Configuration.h.txt](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/11009146/Configuration.h.txt)
[Configuration_adv.h.txt](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/11009147/Configuration_adv.h.txt)

**Comment by radek8** [comment]
The display driver is fine

**Comment by radek8** [comment]
Did you connect the 5-pin cable to the SWD connector on the display or on the SKR board?

**Comment by helakejr** [comment]
5-pin connected to SKR TFT connector.

**Comment by helakejr** [comment]
![connectlcd2](https://user-images.githubusercontent.com/751406/226126458-02240f0e-56a6-4ed9-952d-81a395f8e63b.png)

**Comment by radek8** [comment]
The EXP2 connector is connected incorrectly.
It is shifted by one pin!

![image](https://user-images.githubusercontent.com/46979052/226136669-d5b86ba8-d8cb-458b-ba6e-4f324afcb503.png)

**Comment by radek8** [comment]
Connect the EXP2 connector correctly and hope you didn't destroy anything connected to the SWD connector

**Comment by helakejr** [comment]
Still doesn't work.

**Comment by helakejr** [comment]
![PXL_20230318_210850361 MP](https://user-images.githubusercontent.com/751406/226140450-6a8b48f4-58b3-4612-825a-cf915dc432a7.jpg)
![PXL_20230318_210905292](https://user-images.githubusercontent.com/751406/226140453-b9452e16-7dc8-427c-bd2a-149c117583cb.jpg)
Still doesn't work.

**Comment by helakejr** [comment]
> Did you connect the 5-pin cable to the SWD connector on the display or on the SKR board?
> The other end of the cable remained in the TFT connector?

To get it to work, left TFT 5-pin on correct connector and plugged into the SKR SWD connector.

**Comment by helakejr** [comment]
I don't know if it matters or not but my TFT24 has a GD cpu.

**Comment by radek8** [comment]
> I don't know if it matters or not but my TFT24 has a GD cpu.

CPU type has no effect

**Comment by radek8** [comment]
> > Did you connect the 5-pin cable to the SWD connector on the display or on the SKR board?
> > The other end of the cable remained in the TFT connector?
> 
> To get it to work, left TFT 5-pin on correct connector and plugged into the SKR SWD connector.

How was the connector oriented on the SWD connector?

**Comment by helakejr** [comment]
Reversed from the TFT24 serial port.

**Comment by radek8** [comment]
So you linked together
reset with reset a
5V with 3.3V?

**Comment by helakejr** [comment]
Don't know for sure as I haven't repeated it since your admonishment. But I
believe so.

**Comment by radek8** [comment]
All reset pins are connected together.
By connecting the TFT and SWD connectors, the Reset +5V +3.3V connections were made. The reset and 3.3V buses were powered by 5V.

**Comment by radek8** [comment]
a similar problem

https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/issues/964

**Comment by radek8** [comment]
If touch mode works for you, you don't really need Marlin mode.
I haven't used the Marlin mode on the printer for about 2 years :-)

**Comment by helakejr** [comment]
I believe that this is the solution. Touch mode and its graphical user
interface.

Thank you

**Comment by helakejr** [comment]
Yes

**Comment by helakejr** [comment]
I believe that since Marlin Mode doesn't work and Touch mode is so much
better. I will just use the 5-pin serial connection and save running both
cables.

Thank you for the help.
